File: leetcode_problems/cpp/lc1399_count_largest_group.cpp
#include <algorithm>
#include <iostream>

class Solution
{
public:
    int digit_sum(int n)
    {
        int sum = 0;
        while (n > 0)
        {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }

    int countLargestGroup(int n)
    {
        int arr[50] = {};
        int maxv = 0;
        for (int i = 1; i <= n; i++)
        {
            int ds = digit_sum(i);
            arr[ds]++;
            maxv = std::max(maxv, arr[ds]);
        }

        int res = 0;
        for (int i = 0; i < 50; i++)
        {
            if (arr[i] == maxv)
            {
                res++;
            }
        }
        return res;
    }
};

int main()
{
    Solution sol;
    std::cout << sol.countLargestGroup(10) << std::endl;
    return 0;
}

File: leetcode_problems/typescript/src/lc1399_count_largest_group.ts
function digit_sum(n: number): number {
  let sum = 0;
  while (n > 0) {
    sum += n % 10;
    n = Math.floor(n / 10);
  }
  return sum;
}

function countLargestGroup(n: number): number {
  let arr: number[] = new Array(50).fill(0);
  let maxv: number = 0;
  for (let i = 1; i <= n; i++) {
    let ds = digit_sum(i);
    arr[ds]++;
    maxv = Math.max(maxv, arr[ds]);
  }
  let res = 0;
  for (let i = 0; i < 50; i++) {
    if (arr[i] === maxv) {
      res++;
    }
  }
  return res;
}
